Panic in RangeBackwards

Hi, we are experiencing panic in RangeBackwards function. ttlcache version `v3.3.1-0.202502071402`, includes commit (https://github.com/jellydator/ttlcache/commit/e25850aa352ef9dc3917b9a9224c50304aac3b15) because of deadlocks.

Here is the relevant part of the stacktrace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x714220]

goroutine 8 [running]:
container/list.(*Element).Prev(0xc00018c200?)
	/opt/hostedtoolcache/go/1.23.7/x64/src/container/list/list.go:40
github.com/jellydator/ttlcache/v3.(*Cache[...]).RangeBackwards(0x152d480, 0xc000026e48)
	/home/runner/go/pkg/mod/github.com/jellydator/ttlcache/v3@v3.3.1-0.202502071402[43](https://github.com/e2b-dev/infra/actions/runs/15070929902/job/42366869657#step:6:44)-aefc35918359/cache.go:582 +0xb4
```

--- a/cache.go
+++ b/cache.go
@@ -572,7 +572,7 @@ func (c *Cache[K, V]) Range(fn func(item *Item[K, V]) bool) {
 		return
 	}
 
-	for item := c.items.lru.Front(); item != c.items.lru.Back().Next(); item = item.Next() {
+	for item := c.items.lru.Front(); c.items.lru.Len() != 0 && item != c.items.lru.Back().Next(); item = item.Next() {
 		i := item.Value.(*Item[K, V])
 		expired := i.isExpiredUnsafe()
 		c.items.mu.RUnlock() // unlock mutex so fn func can access it (if it needs to)
@@ -596,7 +596,7 @@ func (c *Cache[K, V]) RangeBackwards(fn func(item *Item[K, V]) bool) {
 		return
 	}
 
-	for item := c.items.lru.Back(); item != c.items.lru.Front().Prev(); item = item.Prev() {
+	for item := c.items.lru.Back(); c.items.lru.Len() != 0 && item != c.items.lru.Front().Prev(); item = item.Prev() {
 		i := item.Value.(*Item[K, V])
 		expired := i.isExpiredUnsafe()
 		c.items.mu.RUnlock() // unlock mutex so fn func can access it (if it needs to)

--- a/cache_test.go
+++ b/cache_test.go
@@ -919,10 +919,28 @@ func Test_Cache_Range(t *testing.T) {
 
 	assert.Equal(t, []string{"5", "4"}, results)
 
+	results = nil
+
+	c.Range(func(item *Item[string, string]) bool {
+		results = append(results, item.Key())
+		return true
+	})
+
+	assert.Equal(t, []string{"5", "4", "3", "2", "1"}, results)
+
 	emptyCache := New[string, string]()
 	assert.NotPanics(t, func() {
 		emptyCache.Range(func(item *Item[string, string]) bool {
 			return false
+		})
+	})
+
+	deletedCache := New[string, string]()
+	addToCache(deletedCache, time.Minute, "6", "3", "4")
+	assert.NotPanics(t, func() {
+		deletedCache.Range(func(item *Item[string, string]) bool {
+			deletedCache.DeleteAll()
+			return true
 		})
 	})
 }
@@ -941,10 +959,28 @@ func Test_Cache_RangeBackwards(t *testing.T) {
 
 	assert.Equal(t, []string{"2", "3", "4"}, results)
 
+	results = nil
+
+	c.RangeBackwards(func(item *Item[string, string]) bool {
+		results = append(results, item.Key())
+		return true
+	})
+
+	assert.Equal(t, []string{"2", "3", "4", "5"}, results)
+
 	emptyCache := New[string, string]()
 	assert.NotPanics(t, func() {
 		emptyCache.RangeBackwards(func(item *Item[string, string]) bool {
 			return false
+		})
+	})
+
+	deletedCache := New[string, string]()
+	addToCache(deletedCache, time.Minute, "6", "3", "4")
+	assert.NotPanics(t, func() {
+		deletedCache.RangeBackwards(func(item *Item[string, string]) bool {
+			deletedCache.DeleteAll()
+			return true
 		})
 	})
 }

